### DISCUSSION TITLE
How to implement the command 'whisper Japanesewav -- language Japanese' in Python code

### DISCUSSION THREAD

**Discussion by JasonChao-yue** [openai/whisper#2439] [Q&A]
How to implement the command 'whisper Japanesewav -- language Japanese' in Python code, with the aim of converting speech into Japanese text

**Comment by agammann** [comment]
Use `model.transcribe()` with `language="ja"` and `task="transcribe"` to turn Japanese speech into Japanese text:

```python
from pathlib import Path
import whisper

model = whisper.load_model("turbo")
result = model.transcribe(
    "Japanese.wav",
    language="ja",
    task="transcribe",
    fp16=(model.device.type == "cuda"),
)

print(result["text"])
Path("Japanese.txt").write_text(result["text"], encoding="utf-8")
```

The corresponding CLI invocation is:

```sh
whisper Japanese.wav --model turbo --language Japanese --task transcribe
```

`ja` is the Japanese language code. `transcribe` preserves the spoken language; `translate` requests translation into English. Use a multilingual model such as `turbo`, `small`, or `base`, rather than an English-only `.en` model. You can change the model name if `turbo` is too large for your machine.

This uses the local `openai-whisper` package (`pip install -U openai-whisper`), and FFmpeg must be installed and available on PATH. The Python call returns a dictionary; the explicit UTF-8 write above saves the text, while the CLI has its own output-file options.

References: [official Python usage and setup](https://github.com/openai/whisper/blob/86098128c0b4f24f0e2aa2994de830614b474227/README.md#python-usage), [language/task handling in transcribe](https://github.com/openai/whisper/blob/86098128c0b4f24f0e2aa2994de830614b474227/whisper/transcribe.py#L143-L165).

AI-assisted answer (Codex), checked against the repository source.
